lib/std/time.zig

```diff
@@ -287,3 +185,250 @@ test "Timer" {
+                    return os.windows.TRUE;
+                }
+            };
+
+            const delta = blk: {
+                // Use 64bit atomics when available to set delta if it hasn't been already
+                if (@sizeOf(usize) >= @sizeOf(u64)) {
+                    const delta = Static.delta.load(.Monotonic);
+                    if (delta != 0) break :blk delta;
+                    break :blk Static.delta.compareAndSwap(
+                        delta,
+                        current,
+                        .Monotonic,
+                        .Monotonic,
+                    ) orelse current;
+                }
+
+                // 64bit atomics aren't supported. Use INIT_ONCE instead
+                var init_with = current;
+                os.windows.InitOnceExecuteOnce(
+                    &Static.delta_once,
+                    Static.initDeltaOnce,
+                    @ptrCast(*c_void, &init_with),
+                    null,
+                );
+                break :blk Static.delta.loadUnchecked();
+            };
+
+            if (current < delta) return 0;
+            return current - delta;
+        }
+
+        /// Calls either GetThreadTimes or GetProcessTimes and returns
+        /// the amount of nanoseconds spent in userspace and the the kernel.
+        fn getCpuTime(comptime GetTimesFn: []const u8, comptime GetCurrentFn: []const u8) !u64 {
+            var creation_time: os.windows.FILETIME = undefined;
+            var exit_time: os.windows.FILETIME = undefined;
+            var kernel_time: os.windows.FILETIME = undefined;
+            var user_time: os.windows.FILETIME = undefined;
+
+            const result = @field(os.windows.kernel32, GetTimesFn)(
+                @field(os.windows.kernel32, GetCurrentFn)(),
+                &creation_time,
+                &exit_time,
+                &kernel_time,
+                &user_time,
+            );
+
+            if (result == 0) {
+                const err = os.windows.kernel32.GetLastError();
+                return os.windows.unexpectedError(err);
+            }
+
+            var cpu_time = (@as(u64, kernel_time.dwHighDateTime) << 32) | kernel_time.dwLowDateTime;
+            cpu_time += (@as(u64, user_time.dwHighDateTime) << 32) | user_time.dwLowDateTime;
+            return cpu_time * 100;
+        }
+    };
+};
+
+test "Clock" {
+    comptime var clock_ids: []const Clock.Id = &[_]Clock.Id{};
+    inline for (std.meta.fields(Clock.Id)) |field| {
+        const clock_id = @field(Clock.Id, field.name);
+        clock_ids = clock_ids ++ [_]Clock.Id{clock_id};
+    }
+
+    for (clock_ids) |clock_id| {
+        _ = Clock.read(clock_id) catch |err| {
+            // Only return errors for clock sources that we use personally.
+            // The rest are just tested for their code paths
+            switch (clock_id) {
+                .realtime, .monotonic => return err,
+                else => continue,
+            }
+        };
+    }
```

Comment:
this logic seems unnecessarily complicated. can it just have the tests instead of a bunch of meta control flow?

Reply:
Using `inline for` on the fields itself made it a compile error to do `continue` for specific errors as it treated it as "runtime-based compile-time control-flow"